NuRadioMC/EvtGen/NuRadioProposal.py

```diff
@@ -88,69 +85,77 @@ def __str__(self):
 calculated by PROPOSAL, and although most of them correspond to actual particles -
 Brems is a bremsstrahlung photon, DeltaE is an ionised electron, and EPair is an
 electron-positron pair, it is useful to treat them as separate entities so that
-we know they come from an interaction. PROPOSAL returns particles as decay products
-only, in our case.
+we know they come from an interaction.
 
 We have followed the PDG recommendation and used numbers between 80 and 89 for
 our own-defined particles, although we have needed also 90 and 91 (they are
 very rarely used).
 """
 
-proposal_interaction_names = { 1000000001: 'particle',
-                               1000000002: 'brems',
-                               1000000003: 'ionized_e',
-                               1000000004: 'e_pair',
-                               1000000005: 'nucl_int',
-                               1000000006: 'mu_pair',
-                               1000000007: 'hadrons',
-                               1000000008: 'cont_loss',
-                               1000000009: 'weak_int',
-                               1000000010: 'compton',
-                               1000000011: 'decay' }
-
-proposal_interaction_codes = { 1000000001: 80,
-                               1000000002: 81,
-                               1000000003: 82,
-                               1000000004: 83,
-                               1000000005: 85,
-                               1000000006: 87,
-                               1000000007: 84,
-                               1000000008: 88,
-                               1000000009: 89,
-                               1000000010: 90,
-                               1000000011: 91 }
-
-
-def particle_code(particle):
+proposal_interaction_names = { pp.particle.Interaction_Type.particle.value: 'particle',
```

Comment:
I experienced a weird bug with these lines.
I tried to run NuRadioMC, and got the error
`AttributeError: 'proposal.particle.Interaction_Type' object has no attribute 'value'`

Took some time to figure this out: 
I have installed PROPOSAL manually (i.e. without conan) on my Ubuntu 18.04 machine.
However, Ubuntu ships with an old pybind11 version (2.4.2, from October 2019), to create the python bindings of PROPOSAL. This version does not include the `value` option for enums, since this has only been included with [pybind11/2.6.2](https://github.com/pybind/pybind11/releases/tag/v2.6.2).

This means that these lines will not work with people who built PROPOSAL with a pybind11 version older than 2.6.2 (this will not be the case for people using pip or conan, but only for users who manually built and installed PROPOSAL with cmake, AND who have an outdated pybind11 version).

What we can do:

1. Keep this with `.value`, which might cause problems in the very-specific case I described above
2. Just go back to how it was before this PR, so explicitly writing the numbers in the code
3. Write `int(pp.particle.Interaction_type.brems)` instead, which appears to be working in both versions
4. ... other ideas?

Reply:
Is this still an issue?

Reply:
I would go for option 3 for the time being and add an appropriate comment

Reply:
I think it is an issue, although a minor one. I would agree with Felix.

Reply:
Agreed? @Jean1995 @shallmann

Reply:
yes, agreed @fschlueter ... are you going to change it or should I? We need to change all the left hand sides in the dict to something like `int(pp.particle.Interaction_type.brems)`, right?

Reply:
can do. Yes, I think this is it.